test/groovy/util/JenkinsReadFileRule.groovy

```diff
@@ -25,9 +32,21 @@ class JenkinsReadFileRule implements TestRule {
             @Override
             void evaluate() throws Throwable {
 
-                testInstance.helper.registerAllowedMethod( 'readFile', [String.class], {s -> return (loadFile("${testRoot}/${s}")).getText('UTF-8')} )
-
-                testInstance.helper.registerAllowedMethod( 'readFile', [Map.class], {m -> return (loadFile("${testRoot}/${m.file}")).getText(m.encoding?m.encoding:'UTF-8')} )
+                testInstance.helper.registerAllowedMethod( 'readFile', [String.class], {s ->
```

Comment:
I don't understand the change here.

Up to now there was an attempt to read a file provided either by `m.file` or by `s` depending on the method was called with a Map or with a String. The file content was delivered from some kind of root directory denoted as `testRoot`.

That should be sufficient also for the new case. Why setting a test file and returning the content of the test file? Why not continuing putting the test file in place under `testRoot`  and returning the content like it was up to now?

What I would understand - and consider as an improvement: changing that rule class to something like register file (path, content) and to deliver the content for a given path in case that path was registered before, but that is something different. In case no file was registered for a given path we should fail with some exception.